rust/private/rustc.bzl

```diff
@@ -675,7 +675,7 @@ def construct_arguments(
     ))
 
     # Set the SYSROOT to the directory of the rust_std files passed to the toolchain
-    env["SYSROOT"] = paths.dirname(toolchain.rust_std.files.to_list()[0].short_path)
+    env["SYSROOT"] = paths.dirname(toolchain.rust_std.to_list()[0].short_path)
```

Comment:
Same here, this could be precomputed once in the toolchain, not recomputed for each action. And while at it, it may be a good idea to check that all the paths in `rust_std` have the same parent directory/or they conform to the sysroot layout rustc expects. The latter thing sounds like a separate PR though.

Reply:
This seems wrong also, I'm pretty sure the sysroot should be a directory, not a path to an arbitrary `rlib` (I think that's what they commonly are). But agreed the toolchain should provide this info.

Reply:
:+1: